pkg/solana/client/multinode/transaction_sender.go

```diff
@@ -94,16 +91,26 @@ type TransactionSender[TX any, CHAIN_ID ID, RPC SendTxRPCClient[TX]] struct {
 // * If there is at least one terminal error - returns terminal error
 // * If there is both success and terminal error - returns success and reports invariant violation
 // * Otherwise, returns any (effectively random) of the errors.
-func (txSender *TransactionSender[TX, CHAIN_ID, RPC]) SendTransaction(ctx context.Context, tx TX) (SendTxReturnCode, error) {
-	txResults := make(chan sendTxResult)
-	txResultsToReport := make(chan sendTxResult)
+func (txSender *TransactionSender[TX, RESULT, CHAIN_ID, RPC]) SendTransaction(ctx context.Context, tx TX) RESULT {
+	txResults := make(chan RESULT)
+	txResultsToReport := make(chan RESULT)
 	primaryNodeWg := sync.WaitGroup{}
 
-	ctx, cancel := txSender.chStop.Ctx(ctx)
-	defer cancel()
+	if txSender.State() != "Started" {
+		return txSender.newResult(errors.New("TransactionSender not started"))
+	}
+
+	txSenderCtx, cancel := txSender.chStop.NewCtx()
+	reportWg := sync.WaitGroup{}
+	defer func() {
+		go func() {
+			reportWg.Wait()
+			cancel()
+		}()
+	}()
 
 	healthyNodesNum := 0
-	err := txSender.multiNode.DoAll(ctx, func(ctx context.Context, rpc RPC, isSendOnly bool) {
+	err := txSender.multiNode.DoAll(txSenderCtx, func(ctx context.Context, rpc RPC, isSendOnly bool) {
```

Comment:
Why can't we pass the standard parent `ctx` here?
If the go routines inside of the func are the problem, then they should handled directly, e.g. via `context.WithoutCancel(ctx)`

Reply:
We only want to cancel this context when transaction sender is closed down. We still need the parent ctx for collecting the results, but the context we sent to multinode is separate.

Reply:
What would happen if you passed `ctx` here, but you used `context.WithoutCancel(ctx)` from the goroutines? Why doesn't that work?

Reply:
If we use `context.WithoutCancel(ctx)` we would be ignoring the TransactionSender being closed inside of the `broadcastTxAsync` and in turn the RPC `SendTx` functions. We want the RPC client to be responsible for the `SendTx` timeout, but still include the case where Transaction Sender if closed.

Reply:
I agree that we should not orphan the goroutines, but this illustrates the point that we can in fact pass the `ctx` normally as a starting point without causing any inherent problems.

Reply:
The only issue is that some RPC calls might timeout slightly early, but this probably won't be an issue. I'll have to run a test to find out.

Reply:
What do you mean that RPC calls might timeout early? There is no need to compromise anything with this approach.

Reply:
> What would happen if you passed ctx here, but you used context.WithoutCancel(ctx) from the goroutines? Why doesn't that work?
@jmank88 if we use `context.WithoutCancel(ctx)` from the goroutines we won't be able to cancel them when txSender is closed.

Reply:
`context.WithCancel` will only disconnect the parent context cancellation. It does not preclude us from cancelling the resulting context. The point is that we _do_ want `multiNode.DoAll` to be responsive to cancellation of the parent context (e.g. don't even launch the goroutines if already cancelled) - only the goroutines workers need to managed separately.

Reply:
Yeah, and we could cancel those internal contexts additional ways if we want/need to.